Leanplum-SDK/Classes/Internal/Leanplum.m

```diff
@@ -510,26 +510,45 @@ + (void)triggerVariablesChangedAndNoDownloadsPending
 + (void)triggerMessageDisplayed:(LPActionContext *)context
 {
     LP_BEGIN_USER_CODE
-    NSString *messageID = context.messageId;
-    NSString *messageKey = @"Message";
-    NSString *messageBody = @"";
-    if ([context.args valueForKey:messageKey]) {
-        messageBody = [context.args valueForKey:messageKey];
-    }
-    NSString *recipientUserID = [Leanplum userId];
-    NSDate *deliveryDateTime = [NSDate date];
+    LPMessageArchiveData *messageArchiveData = [self messageArchiveDataFromContext:context];
     for (LeanplumMessageDisplayedCallbackBlock block in [LPInternalState sharedState]
```

Comment:
Can you help me understand the use of this for loop ?

Reply:
this allows for multiple callback blocks to be added. So users are not restricted to a single callback and are able to add multiple callbacks in different sections of the app. For example, the analytics module could add a callback to analytics and a UI module could add one for UI changes, without having to conflate them both